source/common/network/dns_impl.cc

```diff
@@ -203,8 +216,23 @@ ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name,
   // TODO(hennna): Add DNS caching which will allow testing the edge case of a
   // failed initial call to getHostByName followed by a synchronous IPv4
   // resolution.
+
+  // @see DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback for why this is done.
+  if (dirty_channel_) {
+    ares_destroy(channel_);
+
+    ares_options options{};
+    int optmask = 0;
+
+    if (use_tcp_for_dns_lookups_) {
+      optmask |= ARES_OPT_FLAGS;
+      options.flags |= ARES_FLAG_USEVC;
+    }
```

Comment:
This should be factored into a helper function or moved into `initializeChannel`